flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/PojoSerializerUpgradeTestSpecifications.java

```diff
@@ -229,19 +228,23 @@ public TypeSerializer<StaticSchemaPojo> createUpgradedSerializer() {
             TypeSerializer<StaticSchemaPojo> serializer =
                     TypeExtractor.createTypeInfo(StaticSchemaPojo.class)
                             .createSerializer(new SerializerConfigImpl());
-            assertSame(PojoSerializer.class, serializer.getClass());
+            assertThat(serializer.getClass()).isSameAs(PojoSerializer.class);
             return serializer;
         }
 
         @Override
-        public Matcher<StaticSchemaPojo> testDataMatcher() {
-            return is(new StaticSchemaPojo("Gordon", 27, StaticSchemaPojo.Color.BLUE, false));
+        public Condition<StaticSchemaPojo> testDataCondition() {
+            return new Condition<>(
+                    value ->
+                            new StaticSchemaPojo("Gordon", 27, StaticSchemaPojo.Color.BLUE, false)
+                                    .equals(value),
+                    "");
```

Comment:
Can we provide a meaningful description?
There are many similar cases in this class.